### ISSUE TITLE
Blobstore upload file: Redirect path not found

### ISSUE THREAD

**Issue by sammartens1234** [AppScale/appscale#2523]
Hi,

 I am using docker and started it with -p 8888:8080. I can access my application via {ip-of-machine}:8888/index.html

When I try to upload a blob the following happens: I call the BlobStoreService to create an upload url with a redirect to "/rest/application/blob". The file seems to get uploaded but I get the following error message: 
There was an error with your upload. Redirect path not found. Please contact the app owner if this persists.</br>Server: nginx/1.4.6 (Ubuntu)

Could you help me out on this please,

Sam

**Comment by cdonati** [comment]
It looks like blobstore server is failing to make the specified callback request after adding the blob. I would recommend adding a `logging.exception('Unable to fetch callback URL')` line to the except block on line 399 of AppDB/appscale/datastore/scripts/blobstore.py. After rebuilding the AppDB package with `pip install -U --no-deps /root/appscale/AppDB`, restarting the blobstore server with `monit restart blobstore`, and trying the upload again, you should get a stacktrace in /var/log/appscale/blobstore.log` that will give you more information about what went wrong with the callback request.

**Comment by sammartens1234** [comment]
Solved, I figured it out.

The error I got was MediaType not supported.

On the google appengine I was running my callback url with a MediaType "multipart/form-data". This is where it failed since appscale makes it as "application/x-www-form-urlencoded". If I switch to this my GAE does not run anymore. 
Setting my callback to accept both did the trick. In spring I did the following:

@RequestMapping(value = "blob", method = RequestMethod.POST, consumes = { MediaType.MULTIPART_FORM_DATA_VALUE, MediaType.APPLICATION_FORM_URLENCODED_VALUE })

This was a tricky one to solve. To me it actually makes more sence to have it as "application/x-www-form-urlencoded" since the callback does not upload a file anymore. It is already uploaded.

Hope this helps out some other people. 

Now...on to the next issue ;-)
